### PR TITLE
Cataclysm

### PR DESCRIPTION
**Pull request by Maldaris**
## About The Pull Request

Event game mode for today

## Why It's Good For The Game

https://www.youtube.com/watch?v=Z0GFRcFm-aY

## Changelog
:cl:
add: It's the end of the world as we know it.
/:cl:

### CODE REVIEW

File: waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	light_range = 5
+	light_power = 2
+	light_color="#0066FF"
+	plane = CAMERA_STATIC_PLANE
+
+	var/next_check=0
+	var/list/avail_dirs = list(NORTH,SOUTH,EAST,WEST)
+	var/spread_process_timer
+
+/turf/closed/indestructable/bluespace_cascade/New()
```
Comment (MarkSuckerberg) [inline waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm:15]: any reason this isn't Initialize()?

File: waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	if(!length(avail_dirs))
+		deltimer(spread_process_timer)
+
+	// Choose a direction.
+	var/pdir = pick(avail_dirs)
+	avail_dirs -= pdir
+	var/turf/T=get_step(src,pdir)
+	if(istype(T, /turf/closed/indestructable/bluespace_cascade))
+		avail_dirs -= pdir
+		spread_process_timer = addtimer(CALLBACK(src, .process), 30, TIMER_OVERRIDE | TIMER_STOPPABLE)
```
Comment (MarkSuckerberg) [inline waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm]: I don't think this deletes the old timer, just changes the ID

File: waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		BC.New()
+
+/turf/closed/indestructable/bluespace_cascade/attack_paw(mob/user as mob)
+	return attack_hand(user)
+
+/turf/closed/indestructable/bluespace_cascade/attack_robot(mob/user as mob)
+	if(Adjacent(user))
+		return attack_hand(user)
+	else
+		to_chat(user, "<span class = \"warning\">What the fuck are you doing?</span>")
```
Comment (MarkSuckerberg) [inline waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm:59]: is there a reason this is here but not on normal attack_hand?

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

File: waspstation/code/game/gamemodes/endgame/endgame.dm [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	id = CINEMATIC_BLUESPACE_CATACLYSM
+	is_global = TRUE
+	cleanup_time = 300
+
+/datum/cinematic/bluespace_cataclysm/New()
+	screen = new(src)
+
+/datum/cinematic/bluespace_cataclysm/content()
+	flick("intro_cataclysm", screen)
+	sleep(30)
```
Comment (MarkSuckerberg) [inline waspstation/code/game/gamemodes/endgame/endgame.dm:16]: can this be relegated to a timer?

File: waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm (hunk 3 above)
Comment (Maldaris) [reply waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm:59]: copied from the /vg/ implementation

File: waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm [hunk 5]
```diff
@@ -0,0 +1,107 @@
+// QUALITY COPYPASTA
+/turf/closed/indestructable/bluespace_cascade
+	name = "Bluespace Cascade Foam"
+	desc = "The unimaginable consequences of tampering with Bluespace. The Universe will never be the same."
+	icon='waspstation/icons/turfs/walls/bs_cascade.dmi'
```
Comment (MarkSuckerberg) [inline waspstation/code/game/turfs/closed/indestructable/bluespace_cascade.dm]:
```suggestion
	icon = `waspstation/icons/turf/walls/bs_cascade.dmi`
```

File: waspstation/code/game/gamemodes/endgame/endgame.dm (hunk 4 above)
Comment (Maldaris) [reply waspstation/code/game/gamemodes/endgame/endgame.dm:16]: That's the precedent in other cinematics

**Pull request merged**